### PR TITLE
ci : add building in MSYS2 environments (Windows)

### PR DESCRIPTION
**Pull request by przemoc**
Analogous addition to the one from sister project:

- https://github.com/ggerganov/whisper.cpp/pull/1994

---

windows-msys2 CLANG64 will be failing until #6966 will get merged and this PR rebased then.

### CODE REVIEW

**Review by ggerganov — APPROVED** [review]

**Comment by przemoc** [comment]
Thanks for merging, @ggerganov, but because you merged it before merging #6966 (PR description hinted it should be merged first), you will have one check failing now. And checkers in current PR (possibly due to overly smart patterns in build.yml) did not really build here anything, so that's why you do not see the failure and PR seems green.

**Comment by ggerganov** [comment]
Ah, apologise. Will merge the other PR now and see if everything is good. Thanks

**Pull request merged**
